corehq/apps/userreports/sql/adapter.py

```diff
@@ -260,7 +261,57 @@ def doc_exists(self, doc):
             return session.query(query.exists()).scalar()
 
 
-class ErrorRaisingIndicatorSqlAdapter(IndicatorSqlAdapter):
+class MultiDBSqlAdapter(IndicatorSqlAdapter):
+
+    mirror_adapter_cls = IndicatorSqlAdapter
+
+    def __init__(self, config, override_table_name=None, engine_id=None):
+        from corehq.apps.userreports.models import id_is_static
+        assert id_is_static(config._id)
+        config.validate_db_config()
```

Comment:
I'm not sure if this is the correct place for validation though not completely sure where else it should go. For static data sources it could be called when they are loaded from disk.

Reply:
There's already a validate in `DataSourceConfiguration` https://github.com/dimagi/commcare-hq/blob/04153252e1fe24866018a29c9f58645c77132c01/corehq/apps/userreports/models.py#L467

Reply:
At first I did put it here and then moved for the reason that I thought we only wanted these on static data sources. Since the validate for static data sources doesn't get triggered on save, it didn't make sense to put it in this `validate` method.

Reply:
Removed it from adapter